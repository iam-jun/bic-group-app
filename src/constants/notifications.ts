export const notificationActions = {
  ATTACH: 'attach',
  DETACH: 'detach',
  UPDATE: 'update',
};

export const notificationVerbs = {
  REACT: 'REACT',
  UNREACT: 'UNREACT',
};
