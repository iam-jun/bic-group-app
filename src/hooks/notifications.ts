import { useSelector } from 'react-redux';
import { IObject } from '~/interfaces/common';

const useNotifications = () => {
  const notifications = useSelector(
    (state: IObject<any>) => state.notifications,
  );
  return notifications;
};

export default useNotifications;
