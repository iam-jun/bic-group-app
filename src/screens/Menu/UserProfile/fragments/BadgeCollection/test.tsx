import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
