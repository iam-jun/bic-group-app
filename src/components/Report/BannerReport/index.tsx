import React, { FC } from 'react';
import { View, StyleSheet } from 'react-native';

import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { spacing } from '~/theme';
import Text from '~/baseComponents/Text';
import Icon from '~/baseComponents/Icon';

import { useBaseHook } from '~/hooks';
import usePostsStore from '~/store/entities/posts';
import postsSelector from '~/store/entities/posts/selectors';
import useCommentsStore from '~/store/entities/comments';
import commentsSelector from '~/store/entities/comments/selectors';
import useReportContentStore from '../store';
import { IReportDetail } from '~/interfaces/IReport';
import reportSelector from '../store/selectors';

export interface Props {
  postId?: string;
  commentId?: string;
}

const BannerReport: FC<Props> = ({ postId, commentId }) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = createStyle(theme);
  const { t } = useBaseHook();

  const reportDetailsPost = useReportContentStore(reportSelector.getReportDetailsPost(postId));

  let isReported = null;
  let reportDetails = null;
  if (postId) {
    // Use case Post and Article
    isReported = usePostsStore(postsSelector.getIsReported(postId));
    reportDetails = usePostsStore(postsSelector.getReportDetails(postId)) || reportDetailsPost;
  } else {
    // Use case Comment
    const report = useCommentsStore(commentsSelector.getComment(commentId))?.reportDetails;
    isReported = report?.length > 0;
    reportDetails = report;
  }

  const reportReasons = reportDetails?.map((item: IReportDetail) => item.description).join(', ');

  if (isReported) {
    return (
      <View style={styles.reportContentContainer}>
        <Text.BodyS style={styles.textReportContent}>
          {t('report:banner_report_content')}
          <Text.SubtitleS>{reportReasons}</Text.SubtitleS>
        </Text.BodyS>
        <Icon icon="iconFlagSolid" size={14} tintColor={colors.neutral40} />
      </View>
    );
  }
  return null;
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    reportContentContainer: {
      backgroundColor: colors.neutral10,
      paddingVertical: spacing.padding.small,
      paddingLeft: spacing.padding.large,
      paddingRight: spacing.padding.large + spacing.padding.xSmall,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
    textReportContent: {
      marginRight: spacing.margin.base + spacing.margin.tiny,
    },
  });
};

export default BannerReport;
