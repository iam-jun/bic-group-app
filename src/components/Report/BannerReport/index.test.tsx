import React from 'react';
import usePostsStore from '~/store/entities/posts';

import { renderWithRedux } from '~/test/testUtils';
import BannerReport from './index';
import useCommentsStore from '~/store/entities/comments';
import { NOTI_HIDE_COMMENT, NOTI_HIDE_POST } from '~/test/mock_data/notifications';

describe('BannerReport screen', () => {
  it('render case Post correctly', () => {
    const { actions } = usePostsStore.getState();
    actions.addToPosts({ data: NOTI_HIDE_POST as any, handleComment: true });

    const rendered = renderWithRedux(<BannerReport postId={NOTI_HIDE_POST.id} />);

    expect(rendered).toMatchSnapshot();

    const containerView = rendered.queryByTestId('banner_report');
    expect(containerView).not.toBeNull();
  });

  it('render case Comment correctly', () => {
    const { actions } = useCommentsStore.getState();
    actions.addToComments([NOTI_HIDE_COMMENT]);

    const rendered = renderWithRedux(<BannerReport commentId={NOTI_HIDE_COMMENT.id} />);

    expect(rendered).toMatchSnapshot();

    const containerView = rendered.queryByTestId('banner_report');
    expect(containerView).not.toBeNull();
  });
});
