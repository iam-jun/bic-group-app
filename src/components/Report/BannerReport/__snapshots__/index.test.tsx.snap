// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerReport screen render case Comment correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D6D9EB",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingLeft": 16,
      "paddingRight": 22,
      "paddingVertical": 8,
    }
  }
  testID="banner_report"
>
  <Text
    style={
      Array [
        Object {
          "color": "#212745",
          "fontFamily": "BeVietnamPro-Light",
          "fontSize": 13,
        },
        Object {},
        Object {
          "marginRight": 16,
        },
      ]
    }
  >
    Your content was reported because of 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 13,
          },
          Object {},
          undefined,
        ]
      }
    >
      Spam
    </Text>
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
          Object {
            "backgroundColor": undefined,
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View />
      </View>
    </View>
  </View>
</View>
`;

exports[`BannerReport screen render case Post correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D6D9EB",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingLeft": 16,
      "paddingRight": 22,
      "paddingVertical": 8,
    }
  }
  testID="banner_report"
>
  <Text
    style={
      Array [
        Object {
          "color": "#212745",
          "fontFamily": "BeVietnamPro-Light",
          "fontSize": 13,
        },
        Object {},
        Object {
          "marginRight": 16,
        },
      ]
    }
  >
    Your content was reported because of 
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 13,
          },
          Object {},
          undefined,
        ]
      }
    >
      Illegal, Others, Violent, pornographic, or sexually explicit
    </Text>
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
          Object {
            "backgroundColor": undefined,
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View />
      </View>
    </View>
  </View>
</View>
`;
